File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        
        int n=nums.size();
        int index=0;
        bool isSorted=false;
        for(int i=0;i<n-1;i++)
        {
            if(nums[i+1]>=nums[i])
            isSorted=true;
            else
            {
            isSorted=false;
            break;
            }
        }
        if(isSorted)
        return true;

        for(int i=0;i<n-1;i++)
        {
              if(nums[i+1]<nums[i])

              {
              index=i+1;
              break;
              }
        }
       
       for(int i=index;i<n-1;i++)
       {
         if(nums[i+1]<nums[i])
         return false;
       }
       if(nums[n-1]>nums[0])
       return false;

       for(int i=0;i<index-2;i++)
       {
          if(nums[i+1]<nums[i])
          return false;
       }

       
       return true;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/1878-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        
        int ind=0;
        for(int i=0;i<nums.length-1;i++){
           if(nums[i+1]<nums[i])
           ind=i+1;
        }

        for(int i=ind; i<nums.length-1; i++){
            if(nums[i+1]<nums[i]){
                return false;
            }
        }
        
        if(ind!=0){
            if(nums[nums.length-1]>nums[0])
            return false;
        }

        for(int i=0;i<ind-1;i++){
            if(nums[i]>nums[i+1])
            return false;
        }

        return true;
    }
}